js/ui/codingManager.js

```diff
@@ -927,11 +1035,174 @@ const CodingSession = new Lang.Class({
     }
 });
 
-const CodingManager = new Lang.Class({
-    Name: 'CodingManager',
+// Given a FlatpakInstalledRef app, look up its exports directory
+// and find any relevant .desktop files and add them to the desktop
+// if they are not already added.
+function addFlatpakAppToDesktop(app) {
```

Comment:
I am not sure this function is needed; you should just be able to call `appendIcon` with the app ID. Basically use the same exact behavior the app store has when an app is installed, which ends up here: https://github.com/endlessm/eos-desktop/blob/master/js/ui/shellDBus.js#L493